index.rst

```diff
@@ -1,61 +1,365 @@
+    That request should be routed to some approval process by a manager of the relevant resource, who can then grant or deny the request via the identity management web interface.
+
+IDM-2103
+    Quota grants may expire.
+    The owners of a group must be notified via email of pending quota grant expirations.
+
+Logging
+-------
+
+IDM-2200
+    All group creation, deletion, renaming, and membership changes must be logged.
+
+IDM-2201
+    Owners must be able to see, via the web interface, all history of changes to the group.
+
+IDM-2202
+    All changes to quotas associated with groups must be logged.
+
+IDM-2203
+    Group owners must be able to see a history of all changes to their group quotas.
+
+IDM-2204
+    Users must be able to see a history of all changes to their group membership.
 
-.. TODO: Delete the note below before merging new content to the master branch.
+Administration
+--------------
 
-.. note::
+IDM-2300
+    Administrators must be able to create groups, delete groups, rename groups, and change the membership of any group.
 
-   **This technote is not yet published.**
+IDM-2301
+    Administrators must be able to change the quota grants and requests for any group.
 
-   Requirements for the identity management component of the Science Platform.
+IDM-2302
+    Administrators must be able to impersonate a user and see exactly the same group management and display screens that the user would see.
 
-.. Add content here.
-.. Do not include the document title (it's automatically added from metadata.yaml).
+API
+===
 
-.. .. rubric:: References
+IDM-3000
+    The identity management system must provide a read-only API to other Science Platform components.
+    That API, when given a user authenticator (a token or cookie), must return the user metadata, group memberships, and individual and group quota information.
 
-.. Make in-text citations with: :cite:`bibkey`.
+IDM-3001
+    All actions possible for an administrator to perform in the identity management system must be available via an administrative API as well.
+    This should use separate authentication credentials from user-issued tokens for administrative users.
 
-.. .. bibliography:: local.bib lsstbib/books.bib lsstbib/lsst.bib lsstbib/lsst-dm.bib lsstbib/refs.bib lsstbib/refs_ads.bib
-..    :style: lsst_aa
+IDM-3002
+    Science Platform components may cache the results of read-only API calls to the identity management system, including such information as group membership for a given token and user.
+    The validity of that cache sets a bound on how quickly a token can be revoked.
+    Science Platform components should refresh that information every five minutes, and no less frequently than once per hour.
+    They should not query for the same information from the same token more frequently than every thirty seconds.
+    The identity management system must be able to handle this volume of queries.
```

Comment:
I think you might want to have requirements on the minimum numbers of the following things that need to be supported while maintaining good performance:
* users
* groups
* users in a single group (in practice this probably needs to be as large as the total number of users, or it becomes impossible to do some common things that one might like to do with groups)
* groups associated with a single user

Reply:
I picked some relatively arbitrary scaling limits. Let me know if you would use a different limit than these:

```
IDM-4000
    The identity management system must be able to handle 10,000 active users and 50,000 total users including disabled and frozen users.

IDM-4001
    The identity management system must be able to handle 10,000 active groups and 10,000 members of a single group.

IDM-4002
    The identity management system must be able to retain history of group membership changes for twenty years at a rate of 10 changes per day (100,000 records).

IDM-4003
    The identity management system must allow a single user to be a member of 50 groups.
```